state/accountsDB.go

```diff
@@ -1008,22 +1027,42 @@ func (adb *AccountsDB) SnapshotState(rootHash []byte) {
 	log.Trace("accountsDB.SnapshotState", "root hash", rootHash)
 	trieStorageManager.EnterPruningBufferingMode()
 
+	stats := newSnapshotStatistics(1)
 	go func() {
```

Comment:
We wait for it to finish only if we are in importDB mode (L1044). This needs to be on a goroutine, because if not, the processing will be stoped during this process, and for large tries it can take up to a few minutes.